query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -35,14 +38,83 @@
   // while the selectedColumns
   // is a set of column names.
   private final LinkedHashSet<Expression> allSelections;
+  private final Map<String,String> timeSeriesColumnMap;
+  private final Map<String,List<Long>> timeFilterMap;
 
   public ExecutionContext(String tenantId, QueryRequest request) {
     this.tenantId = tenantId;
     this.selectedColumns = new LinkedHashSet<>();
     this.allSelections = new LinkedHashSet<>();
+    this.timeSeriesColumnMap = new HashMap<>();
+    this.timeFilterMap = new HashMap<String,List<Long>>();
+    analyzeForAvgRate(request);
     analyze(request);
   }
 
+  private void analyzeForAvgRate(QueryRequest request){
```

Comment:
This is a use case, but doesn't belong in here - this shouldn't know who is reading it. How about "scanForTimeFilters" or similar?